cassandra-mesos-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/scheduler/CassandraCluster.java

```diff
@@ -1320,4 +1326,36 @@ static long secondsUntilNextPossibleServerLaunch(final long now, final long next
         }
         return millisUntilNext / 1000;
     }
+
+    public void syncSeedsIfRequired() {
+        long lastTime = clusterState.get().getLastSeedFetchTime();
+        long now = System.currentTimeMillis();
+        if (now - lastTime < TimeUnit.MINUTES.toMillis(1)) return;
+
+        LOGGER.info("Syncing seeds from external DCs ...");
+        List<String> seeds = newArrayList();
+        for (ExternalDc dc : configuration.get().getExternalDcsList())
+            fetchSeeds(dc.getUrl(), seeds);
+
+        LOGGER.info("Got external seeds: " + SEEDS_FORMAT_JOINER.join(seeds));
+        clusterState.updateExternalSeeds(seeds, now);
+    }
+
+    private void fetchSeeds(String url, List<String> seeds) {
+        HttpURLConnection c = null;
+        try {
+            if (url.endsWith("/")) url = url.substring(0, url.length() - 1);
+            url += "/node/seed/all";
+
+            c = (HttpURLConnection)new URL(url).openConnection();
+            JSONObject obj = (JSONObject) new JSONParser().parse(new InputStreamReader(c.getInputStream(), "utf-8"));
+
+            JSONArray seedsArr = (JSONArray) obj.get("seeds");
+            for (Object ip : seedsArr) seeds.add("" + ip);
+        } catch (IOException | ParseException e) {
+            LOGGER.warn("", e);
+        } finally {
+            if (c != null) c.disconnect();
+        }
+    }
```

Comment:
Please move `syncSeedsIfRequired` and `fetchSeeds` into another service class (perhaps `SeedManager` or something). Then pass that new service class into `CassandraCluster` as a constructor argument.

In the new class you can pass in the jackson ObjectMapper to use for json deserialization to a simple struct.